bin/admin-sql

```diff
@@ -1,7 +1,9 @@
 #!/usr/bin/env ruby
 require_relative '../deployment'
 require_relative '../lib/cdo/mysql_console_helper'
+require 'uri'
 
-raise 'please define CDO.db_admin' unless CDO.db_admin
+raise 'please define CDO.db_admin_user' unless CDO.db_admin_user
 
-MysqlConsoleHelper.run(CDO.db_admin, ARGV.join(' ').strip)
+db_connection_settings = URI("mysql://#{CDO.db_admin_user}:#{CDO.db_admin_password}@#{CDO.db_writer_endpoint}:3306/")
```

Comment:
Seems a shame to put all this effort into manually assembling a URI when we've just put all this effort into breaking these values out. What about something like

```ruby
db_connection_settings = URI::Generic.build(
  scheme: 'mysql',
  userinfo: CDO.db_admin_user + ":" + CDO.db_admin_password,
  host: CDO.db_writer_endpoint,
  port: 3306
)
```

Reply:
Agreed. I experimented with that method and found it to be unreliable. `db_connection_settings.to_s` was returning odd results, perhaps because `mysql` is not in the list of schemes supported out of the box? I'll try again.

Reply:
Interesting. It works consistently for me:

```ruby
[development] dashboard > URI("mysql://#{CDO.db_admin_user}:#{CDO.db_admin_password}@#{CDO.db_writer_endpoint}:3306").to_s
=> "mysql://master:pass@writer.com:3306"
[development] dashboard > URI::Generic.build(scheme: 'mysql', userinfo: CDO.db_admin_user + ':' + CDO.db_admin_password, host: CDO.db_writer_endpoint, port: 3306).to_s
=> "mysql://master:pass@writer.com:3306"
```

Although it looks like we do need to specify `path: '/'` if we want to preserve that final `/`:

```ruby
[development] dashboard > URI("mysql://#{CDO.db_admin_user}:#{CDO.db_admin_password}@#{CDO.db_writer_endpoint}:3306/").to_s
=> "mysql://master:pass@writer.com:3306/"
[development] dashboard > URI::Generic.build(scheme: 'mysql', userinfo: CDO.db_admin_user + ':' + CDO.db_admin_password, host: CDO.db_writer_endpoint, port: 3306, path: '/').to_s
=> "mysql://master:pass@writer.com:3306/"
```

Reply:
That's definitely working for me! Switched to that. Much better to construct the URI out of its components instead of just concatenating a bunch of strings together.